components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1681,17 +1681,14 @@ void RewardsServiceImpl::SetAutoContributeEnabled(bool enabled) {
 }
 
 bool RewardsServiceImpl::ShouldShowOnboarding() const {
-  const bool enabled = profile_->GetPrefs()->GetBoolean(prefs::kEnabled);
+  const bool is_enabled = profile_->GetPrefs()->GetBoolean(prefs::kEnabled);
 
-  bool ads_enabled = false;
-  bool ads_supported = true;
-  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile_);
-  if (ads_service) {
-    ads_enabled = ads_service->IsEnabled();
-    ads_supported = ads_service->IsSupportedLocale();
-  }
+  const bool is_ads_enabled =
+      profile_->GetPrefs()->GetBoolean(ads::prefs::kEnabled);
```

Comment:
It seems to me that usage of a particular pref to represent per-profile state is an implementation detail that should be encapsulated by the "component" abstraction that manages it. Is it wise to allow the entire codebase to dig into Ads prefs directly? What happens if we need to change how we store this state? What happens to all of the call sites digging into prefs if the Ads service attaches some non-trivial logic to this storage (e.g. "state migrations")?

Reply:
@zenparsing it's actually just the opposite in general in chromium. There are some "getter" methods in places, but generally speaking prefs are used instead of getter methods because it eliminates a lot of dependencies and startup/shutdown order issues

Reply:
if there is any non-trivial logic around enable/disable, it would be handled by a pref observer in AdsServiceImpl

Reply:
I understand the expediency of directly accessing prefs (most importantly, you get an observation API for free), and I think it could be done in a principled way, where the "public" prefs are clearly documented as such (it seems to me that components would tend to have a mix of implicitly "public" and "private" prefs). Even better, "public" prefs could be exposed in a separate header. Obviously, anything "public" needs to have good documentation as well.

Reply:
I mean, in general I agree with your point, but nothing else in chromium does that so I think we'd just be adding a pattern that confuses people about what to do